07-regression1.Rmd

```diff
@@ -169,24 +181,27 @@ nn_plot <- small_plot +
 nn_plot
 ```
 
-Now that we have the 5 nearest neighbours (in terms of house size) to our new
-2,000 square-foot house of interest, we can use their values to predict a
-selling price for the new home.  Specifically, we can take the mean (or
-average) of these 5 values as our predicted value.
+Figure \@ref(fig:07-knn3-example) illustrates the difference between the house sizes
+of the 5 nearest neighbours (in terms of house size) to our new
+2,000 square-foot house of interest. Now that we have obtained these nearest neighbours, 
+we can use their values to predict the
+sale price for the new home.  Specifically, we can take the mean (or
+average) of these 5 values as our predicted value, as illustrated by
+the red point in Figure \@ref(fig:07-predictedViz-knn).
 
 ```{r 07-predicted-knn}
 prediction <- nearest_neighbours %>%
   summarise(predicted = mean(price))
 prediction
 ```
 
-```{r 07-predictedViz-knn, echo = FALSE, fig.height = 4, fig.width = 5, fig.cap = "Scatter plot of price (USD) versus house size (square footage) with predicted price for a 2000 square-foot house based on 5 nearest neighbours represented as a red dot"}
+```{r 07-predictedViz-knn, echo = FALSE, fig.height = 4, fig.width = 5, fig.cap = "Scatter plot of price (USD) versus house size (square feet) with predicted price for a 2,000 square-foot house based on 5 nearest neighbours represented as a red dot"}
 nn_plot +
   geom_point(aes(x = 2000, y = prediction[[1]]), color = "red", size = 2.5)
 ```
 
-Our predicted price is \$`r format(round(prediction[[1]]), scientific = FALSE)`
-(shown as a red point above), which is much less than \$350,000; perhaps we
+Our predicted price is \$`r format(round(prediction[[1]]), big.mark=",", nsmall=0, scientific = FALSE)`
+(shown as a red point in Figure \@ref(fig:07-predictedViz-knn)), which is much less than \$350,000; perhaps we
 might want to offer less than the list price at which the house is advertised.
 But this is only the very beginning of the story. We still have all the same
 unanswered questions here with K-NN regression that we had with K-NN
```

Comment:
In the section "Training, evaluating, and tuning the model" we say we are going to split the data and put the data in a lock box, then we have a note after that saying we are going to use the entire data set. I think we should swap the order of these two pieces of information to prevent and confusion about what we are using for the training and test sets.